packages/wallet/src/rollup-state-validator.ts

```diff
@@ -0,0 +1,285 @@
+    genesisState: State[],
+    stateMachineDb: DB
+  ): Promise<DefaultRollupStateValidator> {
+    const theRollupMachine = (await DefaultRollupStateMachine.create(
+      genesisState,
+      stateMachineDb,
+      ChecksumAgnosticIdentityVerifier.instance()
+    )) as DefaultRollupStateMachine
+    return new DefaultRollupStateValidator(theRollupMachine)
+  }
+
+  constructor(theRollupMachine: DefaultRollupStateMachine) {
+    this.rollupMachine = theRollupMachine
+  }
+
+  public async getCurrentVerifiedPosition(): Promise<RollupTransitionPosition> {
+    return this.currentPosition
+  }
+
+  public async getInputStateSnapshots(
+    transition: RollupTransition
+  ): Promise<StateSnapshot[]> {
+    if (isSwapTransition(transition)) {
+      const swapperSnapshot: StateSnapshot = await this.rollupMachine.getSnapshotFromSlot(
+        transition.senderSlotIndex
+      )
+      const uniSnapshot: StateSnapshot = await this.rollupMachine.getState(
+        UNISWAP_ADDRESS
+      )
+      return [swapperSnapshot, uniSnapshot]
+    } else if (isCreateAndTransferTransition(transition)) {
+      const nextAccountKey: number = this.rollupMachine.getNextNewAccountSlot()
+      const senderSnapshot: StateSnapshot = await this.rollupMachine.getSnapshotFromSlot(
+        transition.senderSlotIndex
+      )
+      const recipientSnapshot: StateSnapshot = await this.rollupMachine.getSnapshotFromSlot(
+        transition.recipientSlotIndex
+      )
+      return [senderSnapshot, recipientSnapshot]
+    } else if (isTransferTransition(transition)) {
+      const senderSnapshot: StateSnapshot = await this.rollupMachine.getSnapshotFromSlot(
+        transition.senderSlotIndex
+      )
+      const recipientSnapshot: StateSnapshot = await this.rollupMachine.getSnapshotFromSlot(
+        transition.recipientSlotIndex
+      )
+      return [senderSnapshot, recipientSnapshot]
+    }
+  }
+
+  public async getTransactionFromTransitionAndSnapshots(
+    transition: RollupTransition,
+    snapshots: StateSnapshot[]
+  ): Promise<SignedTransaction> {
+    if (isTransferTransition(transition)) {
+      const sender: Address = snapshots[0].state.pubKey
+      const recipient: Address = snapshots[1].state.pubKey
+      const convertedTx: Transfer = {
+        sender,
+        recipient,
+        tokenType: transition.tokenType as UniTokenType | PigiTokenType,
+        amount: transition.amount,
+      }
+      return {
+        signature: transition.signature,
+        transaction: convertedTx,
+      }
+    } else if (isSwapTransition(transition)) {
+      const swapper: Address = snapshots[0].state.pubKey
+      const convertedTx: Swap = {
+        sender: swapper,
+        tokenType: transition.tokenType as UniTokenType | PigiTokenType,
+        inputAmount: transition.inputAmount,
+        minOutputAmount: transition.minOutputAmount,
+        timeout: transition.timeout,
+      }
+      return {
+        signature: transition.signature,
```

Comment:
Will this signature still be valid since it's a signature for a Transition?

Reply:
Yes, confirmed with Karl that the transition is converted to a transaction before sig is checked in contracts.